### PR TITLE
global throwatstation proc

### PR DESCRIPTION
**Pull request by SonixApache**
~~this was supposed to be a draft but whatever~~

adds a global ThrowAtStation proc

~~Feature complete, but when the rod event fires, it throws one (1) runtime that I can't seem to get rid of.~~

### CODE REVIEW

**Comment by CyberpunkBenShapiro** [comment]
The runtime is exactly where and what it says it is. To fix it just get rid of the forcemove proc override for the immovable rod on line 150 it's jank code anyway

**Comment by SonixApache** [comment]
or i can just do that and it'll work™

~~actually i think this just makes the rod invisible but invincible~~

**Comment by CyberpunkBenShapiro** [comment]
the runtime was happening because starting was always null due to the refactor

if(null) code is clutter spaghetti and not a runtime fix

Either make sure the starting variable is defined properly or remove the override proc because an if(null) is the same as commenting it out but more confusing for someone down the line to look at

**Comment by SonixApache** [comment]
I think the runtime was happening because `starting` gets set in immobilerod.dm's `throw_at` proc (L#74), which is normally called before the first `forcemove`.
I'm not sure how to set the var before moving it without more snowflake code.

**Comment by CyberpunkBenShapiro** [comment]
Two ways: 
- make a rod/throwatstation proc that calls throw_at and sets the starting location for forcemove to behave correctly
- realise that the reason why that entire forcemove proc override exists is for a lazy way to check if the z level is the same as the one it spawned in and have it just be deleted when changing z levels

either or

File: code/modules/events/immovablerod.dm [hunk 1]
```diff
@@ -64,10 +42,8 @@
 	projectile_speed = 1.33
 	var/clongSound = 'sound/effects/bang.ogg'
 
-/obj/item/projectile/immovablerod/New(atom/start, atom/end)
+/obj/item/projectile/immovablerod/New(atom/start)
```
Comment (CyberpunkBenShapiro) [inline code/modules/events/immovablerod.dm]: You removed everything in this proc so you can remove this proc too

**Review by CyberpunkBenShapiro — CHANGES_REQUESTED** [review]

**Comment by SonixApache** [comment]
The proc exists to do clang shit, alongside that small z level check.
I think this works; setting the starting var there isn't so bad since it's all local to Z-1 anyway.

**Comment by Blithering** [comment]
This is neat, now we just need a throwstationat proc

**Comment by SGuivenne** [comment]
Could this eventually be used to throw non rod items at station like meteors, cows, or xenomorphs?

**Comment by SonixApache** [comment]
Yes, you just call `ThrowAtStation()` on the meteor, cow or xenomorph and it'll get thrown.

**Pull request merged**